test/new-e2e/tests/npm/helper_test.go

```diff
@@ -0,0 +1,107 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package npm
+
+import (
+	"net"
+	"testing"
+
+	agentmodel "github.com/DataDog/agent-payload/v5/process"
+	krpretty "github.com/kr/pretty"
+
+	"github.com/stretchr/testify/require"
+)
+
+func validateAddr(t *testing.T, addr *agentmodel.Addr) {
+	require.NotEqual(t, 0, len(addr.Ip), "addr.Ip = 0")
+	require.GreaterOrEqualf(t, addr.Port, int32(1), "addr.Port < 1 %d", addr.Port)
+	require.Lessf(t, addr.Port, int32(65536), "addr.Port > 16bits %d", addr.Port)
+
+	require.NotNilf(t, net.ParseIP(addr.Ip), "IP address not valid %s", addr.Ip)
+}
+
+func validateConnection(t *testing.T, c *agentmodel.Connection, cc *agentmodel.CollectorConnections, hostname string) {
+	require.NotZero(t, c.Pid, "Pid = 0")
+	require.NotZero(t, c.NetNS, "network namespace = 0")
+	require.NotNil(t, c.Laddr, "Laddr is nil")
+	require.NotNil(t, c.Raddr, "Raddr is nil")
+
+	validateAddr(t, c.Laddr)
+	validateAddr(t, c.Raddr)
+
+	// un-comment the line below when https://datadoghq.atlassian.net/browse/NPM-2958 will be fixed
+	// require.False(t, c.LastPacketsSent == 0 && c.LastPacketsReceived == 0, "connection with no packets")
+
+	switch c.Type {
+	case agentmodel.ConnectionType_tcp:
+		validateTCPConnection(t, c)
+	case agentmodel.ConnectionType_udp:
+		validateUDPConnection(t, c)
+	}
+
+	validateDNSConnection(t, c, cc, hostname)
+}
+
+func validateDNSConnection(t *testing.T, c *agentmodel.Connection, cc *agentmodel.CollectorConnections, hostname string) {
+	if len(c.DnsStatsByDomainOffsetByQueryType) == 0 {
+		return
+	}
+	for domain, stats := range c.DnsStatsByDomainOffsetByQueryType {
+		for queryType, dnsstat := range stats.DnsStatsByQueryType {
+			domainName, err := cc.GetDNSNameByOffset(domain)
+			require.NoErrorf(t, err, "can't resolve domain tags on %s connection %s", hostname, krpretty.Sprint(c))
+			require.Greaterf(t, len(domainName), 0, "len(domainName) = 0 %s connection %s", hostname, krpretty.Sprint(c))
+			t.Logf("DNS %s query type %v %s", domainName, queryType, krpretty.Sprint(dnsstat))
+			t.Logf("connection to %s:%d", c.Raddr.Ip, c.Raddr.Port)
+		}
+	}
+}
+
+func validateTCPConnection(t *testing.T, c *agentmodel.Connection) {
+	require.Equal(t, c.Type, agentmodel.ConnectionType_tcp, "connection is not TCP")
+	/*
+		The following fields depend on the connection state/scenario
+		 * short term connection
+		 * long term connection
+		 * TCP retransmit packets
+
+		require.NotZero(t, c.LastRetransmits, "LastRetransmits = 0")
+		require.NotZero(t, c.LastTcpClosed, "LastTcpClosed = 0")
+		require.NotZero(t, c.LastTcpEstablished, "LastTcpEstablished = 0")
+	*/
```

Comment:
Do you intend for this comment to remain?

Reply:
I removed the comment and update the require list below 76-78